src/database/DataTypes/PeriodSchedule.js

```diff
@@ -20,12 +20,21 @@ export class PeriodSchedule extends Realm.Object {
    * If the ordertype is an emergency order type, return all periods.
    * @param {MasterList}  program    The related program
    * @param {object}      orderType  The maxOrdersPerPeriod of the orderType
+   * @param {Name}        customer   Optional customer param to filter by.
    */
-  getPeriodsForOrderType(program, orderType) {
+  getPeriodsForOrderType(program, orderType, customer) {
     const { isEmergency } = orderType || {};
 
-    const filterValidPeriods = period =>
-      period.requisitionsForOrderType(program, orderType) < orderType.maxOrdersPerPeriod;
+    const filterValidPeriods = period => {
+      if (customer) {
+        return (
+          period.customerRequisitionsForOrderTypeAndName(program, orderType, customer) <
+          orderType.maxOrdersPerPeriod
+        );
+      }
+
+      return period.requisitionsForOrderType(program, orderType) < orderType.maxOrdersPerPeriod;
```

Comment:
This is finding the 'valid' period for an order type - those which are 'available'.. or have not had 'too many' requisitions - adding an additional customer 'filter', although I don't think this is the best implementation